File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/SEMINAR/S1/untitled/Sem1-MAP/src/MessageTask.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MessageTask extends Task{
    private String message;
    private String from;
    private String to;
    private LocalDateTime date;

    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    @Override
    public String toString() {
        return "id=" + getId() + " | " + "description=" + getDescription() + " | message=" + message + " | from=" +
                from + " | to=" + to + " | date=" + formatter.format(date);
    }

    public MessageTask(String description, String id, String message, String from, String to, LocalDateTime date) {
        super(description, id);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    @Override
    public void execute() {
        System.out.println(date.format(formatter)+": "+message);
    }
}

File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 11 - homework/lab11/lab11/task/MessageTask.cs
using System.Transactions;

namespace lab11;

public class MessageTask : Task {
    private string message;
    private string from;
    private DateTime date;

    public MessageTask(string from, string message, DateTime date, string taskID, string description):base(taskID, description)
    {
        this.message = message;
        this.from = from;
        this.date = date;
    }
    
    public override void Execute()
    {
        Console.WriteLine(TaskID+" "+Description+" "+from+" "+message+" "+date);
    }

    public string Message
    {
        get { return message; }
        set { message = value; }
    }

    public string From
    {
        get { return from; }
        set { from = value; }
    }

    public DateTime Date
    {
        get { return date; }
        set { date = value; }
    }
}